data/libs/SpaceStation.lua

```diff
@@ -35,43 +35,155 @@ function SpaceStation:Constructor()
+
+	local pricemod = Game.system:GetCommodityBasePriceAlterations(key)
+	local targetStock = rn * (rand:Number() + rand:Number()) / 2.0 -- normal 0-100% "permanent" stock
+	return applyStockPriceMod(rn, targetStock, pricemod)
+end
+
+-- create a persistent entry for the given station's commodity market if it
+-- does not already exist, and populate persistent and transient stock info
+local function createStationMarket(station)
 	assert(station and station:exists())
-	if equipmentStock[station] then return end
-	equipmentStock[station] = {}
+	if stationMarket[station.path] then return end
 
-	local hydrogen = Equipment.cargo.hydrogen
-	for key, e in pairs(Equipment.cargo) do
-		if e.purchasable then
-			local rn = 100000 / math.abs(e.price) --have about 100,000 worth of stock, per commodity
-			if e == hydrogen then
-				equipmentStock[station][e] = math.floor(rn/2 + Engine.rand:Integer(0,rn)) --always stock hydrogen
-			else
-				local pricemod = Game.system:GetCommodityBasePriceAlterations(key)
-				local stock =  (Engine.rand:Integer(0,rn) + Engine.rand:Integer(0,rn)) / 2 -- normal 0-100% stock
-				if pricemod > 10 then --major import, low stock
-					stock = stock - (rn*0.10)     -- shifting .10 = 2% chance of 0 stock
-				elseif pricemod > 4 then --minor import
-					stock = stock - (rn*0.07)     -- shifting .07 = 1% chance of 0 stock
-				elseif pricemod < -10 then --major export
-					stock = stock + (rn*0.8)
-				elseif pricemod < -4 then --minor export
-					stock = stock + (rn*0.3)
-				end
-				equipmentStock[station][e] = math.floor(stock >=0 and stock or 0)
-			end
-		else
-			equipmentStock[station][e] = 0 -- commodity that cant be bought
+	local storedStation = {
+		commodities = {},
+		lastStockUpdate = Game.time
+	}
+	stationMarket[station.path] = storedStation
+
+	local h2 = Equipment.cargo.hydrogen
+	for key, e in pairs (Equipment.cargo) do
+		if e.purchasable and e.price > 0.0 and e ~= h2 then
+			local targetStock = SpaceStation.GetStationTargetStock(key, station.seed)
+			storedStation.commodities[key] = targetStock
+			equipmentStock[station][e] = targetStock
 		end
 	end
+end
+
+-- handle gradually restocking commodities at a station over time
+-- by default, a station restores to its maximum stock after 12 weeks
+-- this function is safe to call at any time, though it should be rate-limited
+local kTickDuration = 7 * 24 * 60 * 60 -- 1 week
+local kAvgTicksToRestock = 12
+local function updateStationMarket (station)
+	assert(station and station:exists())
+	if not stationMarket[station.path] then return end
+
+	local storedStation = stationMarket[station.path]
+	local lastStockUpdate = storedStation.lastStockUpdate
+	local timeSinceUpdate = Game.time - lastStockUpdate
+	if timeSinceUpdate <= kTickDuration then return end
+
+	-- make sure the next tick happens at the correct time
+	storedStation.lastStockUpdate = lastStockUpdate + math.floor(timeSinceUpdate / kTickDuration) * kTickDuration
```

Comment:
nevermind, I was stupid

Reply:
This line in particular is meant to make sure that updates happen regularly every single week since start, even if the player jumps back into the system a week and a half later and there's half a week of time that isn't being processed.